### PR TITLE
For #7702 - Fix nightly start up test (WIP)

### LINKED ISSUE #7702: [Bug] Fix Nightly androidTest apk for startup test
## Summary
The nightly startup test: [NavigationToolbarTest#visitUrlTest](https://github.com/mozilla-mobile/fenix/blob/master/app/src/androidTest/java/org/mozilla/fenix/ui/NavigationToolbarTest.kt#L117-L125)  is a gating UI test that is intended to run just prior to publishing to playstore.  If test fails, app crashes, etc., nightly should not be published.  Test is currently disabled because it perma-fails on Firebase (and locally from command line) even though the very same test works 100% of the time on debug builds on github events.

## git PR or merge to master
When submitting a PR or merging to master, both the (debug) app and androidTest apks are built and the UI tests pass.  The package names from the apks in the taskcluster payload are identical:
`org.mozilla.fenix.debug` and
`org.mozilla.fenix.debug.test` respectively

## Nightly startup test
When the same test is run prior to publishing Nightly, both the (nightly) app and androidTest apks are built and the UI test fails.  The package names from the apks in the taskcluster payload are different:
`org.mozilla.fenix.nightly` and
`org.mozilla.fenix.debug.test`

when trying to run the tests locally from the command line these apks produces the following error....

```
INSTRUMENTATION_STATUS: Error=Unable to find instrumentation
```

## Notes
**get apks**
wget apks from [taskcluster payload](https://firefox-ci-tc.services.mozilla.com/tasks/MRGKqXFaR5mUs0TYp_57AQ)

**get package name**
```
$ ~/Library/Android/sdk/build-tools/29.0.2/aapt dump badging app.apk  | grep package:\ name
package: name='org.mozilla.fenix.debug' versionCode='1' versionName='1.0.2003' compileSdkVersion='28' compileSdkVersionCodename='9'
```
**run test w/ pre-installed apks**
```
adb shell am instrument -w -r -e debug false -e class org.mozilla.fenix.ui.NavigationToolbarTest#visitURLTest org.mozilla.fenix.debug.test/androidx.test.runner.AndroidJUnitRunner
```

### PR DESCRIPTION
**Pull request by isabelrios**
Fixes #7702 Still in early stage but opening to share progress and and questions about it  since there are some modifications affecting the app that need to be fixed in a  different way

### CODE REVIEW

**Comment by isabelrios** [comment]
This was just to try a solution, changes will be added in PR #10152

**Comment by isabelrios** [comment]
closed by error, reopening

File: app/src/androidTest/java/org/mozilla/fenix/helpers/MockWebServer.kt [hunk 1]
```diff
@@ -13,7 +13,7 @@ import okhttp3.mockwebserver.MockResponse
 import okhttp3.mockwebserver.MockWebServer
 import okhttp3.mockwebserver.RecordedRequest
 import okio.Buffer
-import okio.source
+//import okio.source
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/helpers/MockWebServer.kt:16]: I needed to make some changes in this file to get the build to compile, not correct but focusing on getting the apk and coming back to fix these errors later

**Comment by liuche** [comment]
Is this PR still in progress @isabelrios?

**Comment by gabrielluong** [comment]
Hi @isabelrios, I am closing this PR for now since it has gone a bit stale and to help clean up the PR backlog for now. We can always reopen it.

**Pull request closed without merging**